### PR TITLE
Add frost free season indicators [rebase]

### LINKED ISSUE #887: Indicator request: growing_season_start and frost_free_season indicators
Using the `xclim.atmos.freshet_start`, offer a `growiing_season_start` indicator. The same can be done for indicators of `frost_free_season`.

### PR DESCRIPTION
**Pull request by Zeitsperre**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #887 
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added.
- [x] `bumpversion patch` has been called on this branch
- [ ] The relevant author information has been added to `.zenodo.json`

### What kind of change does this PR introduce?

* adds a few indicators with metadata specific to frost free season
* add `growing_season_start` Indicator based on `freshet_start` indice
* add thawing_degree_days
* add freezing_degree_days
* add French translations for new indicators
* Correct usage of `"."` in `description`, `long_name`, and `title` strings

### Does this PR introduce a breaking change?

I've added a default suggested value to some growing_degree_day calculations. This might affect some users. If this should be undone, let me know.

### Other information:

~There's an identical indice proposed here, solely with differences in metadata (`freshet_start` = `frost_free_season_start`). I wanted to propose this since "freshet" is not as commonly used in literature as "frost free season". If this ruffles feathers, I can remove it.~ This indicator needed to be fully implemented. Can't cut corners.

### CODE REVIEW

File: xclim/indicators/atmos/_temperature.py [hunk 1]
```diff
@@ -607,11 +624,49 @@ class Temp(Daily):
...
     compute=indices.maximum_consecutive_frost_days,
 )
 
+frost_free_season_length = Temp(
+    identifier="frost_free_season_length",
+    units="days",
+    standard_name="days_with_air_temperature_above_threshold",
+    long_name="Length of the frost free season",
+    description="{freq} number of days between the first occurrence of at least "
+    "{window} consecutive days with minimum daily temperature above freezing and "
```
[the remark this answers is not in the document]
Comment (Zeitsperre) [reply xclim/indicators/atmos/_temperature.py:638]: I just realized we literally have a `frost_season_length` indice. Modifying that now.

[the remark this answers is not in the document]
Comment (Zeitsperre) [reply xclim/indicators/atmos/_temperature.py:638]: It looks as though `indices.growing_season_length` already is using `tas >= thresh` so the `frost_free_season` indicator is in the clear.

File: xclim/indicators/atmos/_temperature.py [hunk 2]
```diff
@@ -607,11 +624,49 @@ class Temp(Daily):
...
+    "the first occurrence of at least {window} consecutive days with "
+    "minimuim daily temperature below freezing after {mid_date}.",
+    cell_methods="time: minimum within days time: sum over days",
+    compute=indices.growing_season_length,
+    parameters={"thresh": {"default": "0 degC"}},
+)
+
+frost_free_season_start = Temp(
```
[the remark this answers is not in the document]
Comment (Zeitsperre) [reply xclim/indicators/atmos/_temperature.py:646]: Good question. They return different values, so we should probably continue offering both. Given that they're so closely related, it makes a good argument for dismantling some of these indices into generic terms (e.g. `season_start` and `season_end`).

**Comment by Zeitsperre** [comment]
@juliettelavoie I've added a `growing_season_start` indice to fix the issues you mentioned. Thanks for the review comment.

This PR is good for final review IMO.

File: xclim/indicators/atmos/_temperature.py [hunk 3]
```diff
@@ -607,11 +624,49 @@ class Temp(Daily):
...
 
+frost_free_season_length = Temp(
+    identifier="frost_free_season_length",
+    units="days",
+    standard_name="days_with_air_temperature_above_threshold",
+    long_name="Length of the frost free season",
+    description="{freq} number of days between the first occurrence of at least "
+    "{window} consecutive days with minimum daily temperature above freezing and "
+    "the first occurrence of at least {window} consecutive days with "
+    "minimuim daily temperature below freezing after {mid_date}.",
```
Comment (juliettelavoie) [inline xclim/indicators/atmos/_temperature.py:640]:
```suggestion
    "minimum daily temperature below freezing after {mid_date}.",
```

File: xclim/data/fr.json [hunk 4]
```diff
@@ -483,25 +483,32 @@
...
     "long_name": "Degré-jour de croissance (Tmoy > {thresh})",
-    "description": "Cumul {freq} des degrés-jours de croissance (Température au-dessus de {thresh})",
+    "description": "Cumul {freq} des degrés-jours de croissance (Température au-dessus de {thresh}).",
     "title": "Degré-jour de croissance",
     "comment": "",
     "abstract": "Cumul des degrés-jours pour les jours où la température moyenne est au-dessus d'un seuil."
   },
+  "THAWING_DEGREE_DAYS": {
+    "long_name": "Degré-jour de dégel (Tmoy > {thresh})",
+    "description": "Cumul {freq} des degrés-jours de dégel(Température au-dessus de {thresh}).",
```
Comment (aulemahal) [inline xclim/data/fr.json:507]:
```suggestion
    "description": "Cumul {freq} des degrés-jours de dégel (Température au-dessus de {thresh}).",
```

File: xclim/data/fr.json [hunk 5]
```diff
@@ -674,7 +702,14 @@
...
+    "abstract": "Jour de l'année de la fin de la saison de végétation. La saison de végétation est définie comme l'intervalle entre la première série de N jours où la température journalière est au-dessus d'un seuil et la première série (après une certaine date) de N jours où elle est sous ce même seuil."
+  },
+  "FROST_FREE_SEASON_END": {
+    "long_name": "Jour de l'année de la fin de la saison sans gel",
+    "description": "Jour de l'année de la fin de la saison sans gel. La saison sans gél est définie comme l'intervalle entre la première série de {window} jours où la température journalière est au-dessus de {thresh} et la première série (après le {mid_date}) de {window} jours où elle est sous {thresh}.",
+    "comment": "",
+    "title": "Fin de la saison sans gél",
+    "abstract": "Jour de l'année de la fin de la saison sans gél. La saison sans gél est définie comme l'intervalle entre la première série de N jours où la température journalière est au-dessus d'un seuil et la première série (après une certaine date) de N jours où elle est sous ce même seuil."
```
Comment (aulemahal) [inline xclim/data/fr.json:712]:
```suggestion
    "description": "Jour de l'année de la fin de la saison sans gel. La saison sans gel est définie comme l'intervalle entre la première série de {window} jours où la température journalière est au-dessus de {thresh} et la première série (après le {mid_date}) de {window} jours où elle est sous {thresh}.",
    "comment": "",
    "title": "Fin de la saison sans gel",
    "abstract": "Jour de l'année de la fin de la saison sans gél. La saison sans gel est définie comme l'intervalle entre la première série de N jours où la température journalière est au-dessus d'un seuil et la première série (après une certaine date) de N jours où elle est sous ce même seuil."
```

Comment (aulemahal) [reply xclim/data/fr.json:712]: The texts for `frost_free_season_start` and  `frost_free_season_end` are quite different, is there a reason for this?

File: xclim/indicators/atmos/_temperature.py (hunk 1 above)
Comment (juliettelavoie) [inline xclim/indicators/atmos/_temperature.py:638]:
```suggestion
    "{window} consecutive days with minimum daily temperature above or at the freezing point and "
```

File: xclim/indicators/atmos/_temperature.py [hunk 6]
```diff
@@ -607,11 +624,49 @@ class Temp(Daily):
...
+    parameters={"thresh": {"default": "0 degC"}},
+)
+
+frost_free_season_start = Temp(
+    identifier="frost_free_season_start",
+    units="",
+    standard_name="day_of_year",
+    long_name="Day of year of frost free season start",
+    description="Day of year of beginning of frost free season, defined as the first day a temperature "
+    "threshold of {thresh} is exceeded for at least {window} days.",
```
Comment (juliettelavoie) [inline xclim/indicators/atmos/_temperature.py:652]:
```suggestion
    "threshold of {thresh} is equal or exceeded for at least {window} days.",
```

**Review by aulemahal — APPROVED** [review]
Nice addition, thanks!

File: xclim/indicators/atmos/_temperature.py [hunk 7]
```diff
@@ -623,6 +678,19 @@ class Temp(Daily):
     compute=indices.maximum_consecutive_frost_free_days,
 )
 
+growing_season_start = Temp(
+    identifier="growing_season_start",
+    units="",
+    standard_name="day_of_year",
+    long_name="Day of year of growing season start",
+    description="Day of year of start of growing season, defined as the first day of "
+    "consistent superior threshold temperature of {thresh} after a run of "
```
Comment (juliettelavoie) [inline xclim/indicators/atmos/_temperature.py:687]:
```suggestion
    "consistent superior or equal to threshold temperature of {thresh} after a run of "
```

File: xclim/data/fr.json (hunk 5 above)
Comment (Zeitsperre) [reply xclim/data/fr.json:712]: I think I based them on existing indices, I can try and synchronize them for good measure.

File: xclim/indices/_threshold.py [hunk 8]
```diff
@@ -555,6 +556,51 @@ def growing_degree_days(
...
 
 
+@declare_units(tas="[temperature]", thresh="[temperature]")
+def growing_season_start(
+    tas: xarray.DataArray, thresh: str = "5.0 degC", window: int = 5, freq: str = "YS"
+) -> xarray.DataArray:
+    r"""Start of the growing season.
+
+    Day of the year of the start of a sequence of days with a temperature consistently
+    above a threshold, after a period with temperatures consistently above the same threshold.
```
Comment (juliettelavoie) [inline xclim/indices/_threshold.py:566]:
```suggestion
    above or equal a threshold, after a period with temperatures consistently above the same threshold.
 ```

**Review by juliettelavoie — APPROVED** [review]

File: xclim/indices/_threshold.py (hunk 8 above)
Comment (Zeitsperre) [reply xclim/indices/_threshold.py:566]:
```suggestion
    above or equal to a threshold, after a period with temperatures consistently above the same threshold.
```

**Comment by Zeitsperre** [comment]
Added `freezing_degree_days` since well, why not. Thanks for the reviews both of you!

**Comment by juliettelavoie** [comment]
I just realized that `growing_season_lenght` used `tas`, not tasmin as described and as would make sense for `frost_free_season_length`.

**Comment by Zeitsperre** [comment]
> I just realized that `growing_season_length` used `tas`, not `tasmin` as described in the metadata of `frost_free_season_length`. `frost_season_length` uses `tasmin`.
> 
> (same for _start and _end)

Right. I really shouldn't be rushing this PR through like this. Will make the needed modifications.

**Comment by juliettelavoie** [comment]
> > I just realized that `growing_season_length` used `tas`, not `tasmin` as described in the metadata of `frost_free_season_length`. `frost_season_length` uses `tasmin`.
> > (same for _start and _end)
> 
> Right. I really shouldn't be rushing this PR through like this. Will make the needed modifications.

and I think it makes sense that `frost_free_season_length` should use `tasmin`, so I don't know that we can even use `growing_season_length` ( or maybe modify it to be more flexible to be able to choose btw `tas` and `tasmin`)

**Comment by Zeitsperre** [comment]
> and I think it makes sense that `frost_free_season_length` should use `tasmin`, so I don't know that we can even use `growing_season_length` ( or maybe modify it to be more flexible to be able to choose btw `tas` and `tasmin`)

We can't leverage `growing_season_*` indides, no, The good news is this will correct some underlying data. It's just a bit more code.

**Comment by aulemahal** [comment]
Since #873, you can use `input={'tasmin': 'tas'}` to switch a variable in the compute's signature. Could that be useful here?

As for the flexibility question, we have no current way of making this. But maybe we should.

**Pull request merged**